tests/benchmark/crud_test.go

```diff
@@ -0,0 +1,920 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package benchmark
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+func getRootResource(b *testing.B) string {
+	b.Helper()
+
+	p, ok := os.LookupEnv("REGISTRY_PROJECT_IDENTIFIER")
+	if !ok || p == "" {
+		p = "bench"
+	}
+	return fmt.Sprintf("projects/%s/locations/global", p)
+}
+
+func loadVersionCount(b *testing.B) int {
+	b.Helper()
+
+	s, ok := os.LookupEnv("REGISTRY_VERSION_COUNT")
+	if !ok {
+		return 1
+	}
+
+	c, err := strconv.Atoi(s)
+	if err != nil {
+		b.Fatalf("Setup: Invalid REGISTRY_VERSION_COUNT %q: count must be an integer", s)
+	}
+
+	return c
+}
+
+func getApiName(i int, b *testing.B) string {
+	b.Helper()
+
+	prefix, ok := os.LookupEnv("API_NAME_PREFIX")
+	if !ok {
+		prefix = "benchtest"
+	}
+	s, ok := os.LookupEnv("API_NAME_START_OFFSET")
+	if ok {
+		offset, err := strconv.Atoi(s)
+		if err != nil {
+			b.Fatalf("Setup: Invalid API_NAME_START_OFFSET %q: offset must be an integer", s)
+		}
+		i += offset
+	}
+
+	return fmt.Sprintf("%s-%d", prefix, i)
```

Comment:
>Say I ran the create script and created 100 APIs. I wanted to add 100 more APIs to an existing environment without having to delete and rerun the script with 200 APIs. This was Needed when I started seeing errors while trying to adding 2000 APIs. so I could break down the create into smaller chunks

I understand it was helpful during development, but why are we including it now? I'm not understanding the benefit to having custom prefixes _and_ offsets. If I want to add 100 more APIs, then I can pick a new prefix and re-run the benchmarks on the same project.

Reply:
Say I want to create 10k APIs to test.
I can run the create scripts 5 times with 2k APIs created every time.
Once the 10k APIs are created, we can now run the GET and LIST benchmarks on the 10k APIs and they follow the same pattern for api names (test-1000 test-5000 , test-10000)